akvo/rsr/tests/usecases/test_previous_cumulative_update_by_user.py

```diff
@@ -0,0 +1,253 @@
+
+
+class NonCumulativeIndicatorTestCase(BaseTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.user = self.create_user('test@akvo.org', 'password', is_admin=True)
+        self.project = ProjectFixtureBuilder()\
+            .with_disaggregations({
+                'Gender': ['Male', 'Female']
+            })\
+            .with_results([
+                {
+                    'title': 'Result #1',
+                    'indicators': [
+                        {
+                            'title': 'Indicator #1',
+                            'periods': [
+                                {
+                                    'period_start': date(2010, 1, 1),
+                                    'period_end': date(2010, 12, 31),
+                                },
+                                {
+                                    'period_start': date(2011, 1, 1),
+                                    'period_end': date(2011, 12, 31),
+                                },
+                            ]
+                        }
+                    ]
+                }
+            ])\
+            .build()
+        self.indicator = self.project.indicators.get(title='Indicator #1')
+
+    def test_no_cumulative_update(self):
+        self.project.get_period(period_start=date(2010, 1, 1)).add_update(
+            self.user, value=10, disaggregations={
+                'Gender': {
+                    'Male': {'value': 3},
+                    'Female': {'value': 7},
+                }
+            }
+        )
+        result = get_previous_cumulative_update_value(self.user, self.indicator)
+        self.assertEqual(result, empty_result())
+
+
+class CumulativeUnitIndicatorUserUpdateTestCase(BaseTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.user = self.create_user('test_user@akvo.org', 'password', is_admin=True)
+        self.project = ProjectFixtureBuilder()\
+            .with_disaggregations({
+                'Gender': ['Male', 'Female']
+            })\
+            .with_results([
+                {
+                    'title': 'Result #1',
+                    'indicators': [
+                        {
+                            'title': 'Indicator #1',
+                            'cumulative': True,
+                            'periods': [
+                                {
+                                    'period_start': date(2010, 1, 1),
+                                    'period_end': date(2010, 12, 31),
+                                },
+                                {
+                                    'period_start': date(2011, 1, 1),
+                                    'period_end': date(2011, 12, 31),
+                                },
+                            ]
+                        }
+                    ]
+                }
+            ])\
+            .build()
+        self.indicator = self.project.indicators.get(title='Indicator #1')
```

Comment:
This seems to be the same in all added classes. A base class for cumulative updates could be used.

Reply:
fixed